dns_sd_windows.c

```diff
@@ -19,42 +19,60 @@
 #include "iio-private.h"
 #include "mdns.h"
 
-static int new_discovery_data(struct dns_sd_discovery_data** data)
+#ifdef HAVE_IPV6
+static const unsigned char localhost[] = {
+	0, 0, 0, 0, 0, 0, 0, 0,
+	0, 0, 0, 0, 0, 0, 0, 1
+};
+static const unsigned char localhost_mapped[] = {
+	0, 0, 0,    0,    0,    0, 0, 0,
+	0, 0, 0xff, 0xff, 0x7f, 0, 0, 1
+};
+
+static inline bool is_localhost6(struct sockaddr_in6 *saddr6)
```

Comment:
ditto for the ``const?

More like a nitpick, in general I'm not a big fan of the `inline` (even being it a static function that only appears to be used once) as the compiler is smart enough to do it already without the hint. It can even be more "smarter" and ignore the keyword at all :smile: (which I don't think it would do on this case)...

Anyways, just my 2 cents

Reply:
The thing with `static inline` is that you can have it in a header, the compiler won't complain about it being defined everywhere, and won't complain if it's not used in a particular source file. I'm just too used to write `static inline` when I write functions that could be macros, there's not more to it.

Reply:
> The thing with static inline is that you can have it in a header

In headers sure... no way around it